crates/astria-conductor/src/reader.rs

```diff
@@ -282,3 +351,249 @@ impl Reader {
         Ok(())
     }
 }
+
+/// This function ensures that the given Commit has quorum, ie that the Commit contains >2/3 voting
+/// power. It performs the following checks:
+/// - the height of the commit matches the block height of the validator set
+/// - each validator in the commit is in the validator set
+/// - for each signature in the commit, the validator public key matches the validator address in
+///   the commit
+/// - for each signature in the commit, the validator signature in the commit is valid
+/// - the total voting power of the commit is >2/3 of the total voting power of the validator set
+///
+/// # Errors
+///
+/// If any of the above conditions are not satisfied, an error is returned.
+#[instrument]
+fn ensure_commit_has_quorum(
+    commit: &Commit,
+    validator_set: &ValidatorSet,
+    chain_id: &str,
+) -> Result<()> {
+    if commit.height != validator_set.block_height {
+        bail!(
+            "commit height mismatch: expected {}, got {}",
+            validator_set.block_height,
+            commit.height
+        );
+    }
+
+    let mut total_voting_power = 0u64;
```

Comment:
Do we need to be careful if the voting powers are too large, i.e. if their orders of magnitude are in the vicinity of u64::MAX? What protections do we have in place to guard against that?

Reply:
I don't think that can happen, we are also receiving these values from a sequencer client that we're running or that we trust, so there should not be invalid values. however I'll double check what the values can be

Reply:
I would very much like to have some kind of early warning system in place just in case we f up something on the tendermint. It's true that we work with trusted inputs, but that does not mean that we should trust our inputs. :smile: 

The reason I am flagging this part is that it can silently break without us noticing.

What do you think about using `checked_add`?:

```rust
let Some(total_voting_power) = validator_set
    .validators
    .iter()
    .try_fold(0, |acc, &power| acc.checked_add(power)) else {
    bail!("total voting power exceeded u64:MAX");
};
```